src/Mod/Arch/ArchRoof.py

```diff
@@ -286,393 +383,455 @@ def makeRoofProfilsDic(self, id, angle, run, idrel, overhang, thickness,):
 
-    def createProfilShape (self, points, midpoint, rot, vec, run, d, f):
+        profilCurr["points"] = self.ptsPaneProject
 
-        import Part
+    def createProfilShape (self, points, midpoint, rot, vec, run, diag, sol):
         lp = len(points)
         points.append(points[0])
         edgesWire = []
         for i in range(lp):
-            edge = Part.makeLine(points[i],points[i+1])
+            edge = Part.makeLine(points[i],points[i + 1])
             edgesWire.append(edge)
         profil = Part.Wire(edgesWire)
         profil.translate(midpoint)
-        profil.rotate(midpoint,FreeCAD.Vector(0,0,1), 90. + rot * -1)
-        perp = self.getPerpendicular(vec,rot,run)
-        profil.rotate(midpoint,perp,90.)
+        profil.rotate(midpoint, Vector(0.0, 0.0, 1.0), 90.0 - rot)
+        per = self.getPerpendicular(vec, rot, run)
+        profil.rotate(midpoint, per, 90.0)
         vecT = vec.normalize()
-        vecT.multiply(d)
+        vecT.multiply(diag)
         profil.translate(vecT)
-        vecE = vecT.multiply(-2.)
+        vecE = vecT.multiply(-2.0)
         profilFace = Part.Face(profil)
         profilShp = profilFace.extrude(vecE)
-        profilShp = f.common(profilShp)
+        profilShp = sol.common(profilShp)
         #shapesList.append(profilShp)
         return profilShp
 
-    def execute(self,obj):
+    def execute(self, obj):
 
         if self.clone(obj):
             return
 
-        import Part, math, DraftGeomUtils
         pl = obj.Placement
         #self.baseface = None
         self.flip = False
-        if hasattr(obj,"Flip"):
+        if hasattr(obj, "Flip"):
             if obj.Flip:
                 self.flip = True
         base = None
-        w = None
+        baseWire = None
         if obj.Base:
-            if hasattr(obj.Base,'Shape'):
+            if hasattr(obj.Base, "Shape"):
                 if obj.Base.Shape.Solids:
                     base = obj.Base.Shape
                     #pl = obj.Base.Placement
                 else:
                     if (obj.Base.Shape.Faces and obj.Face):
-                        w = obj.Base.Shape.Faces[obj.Face-1].Wires[0]
+                        baseWire = obj.Base.Shape.Faces[obj.Face-1].Wires[0]
                     elif obj.Base.Shape.Wires:
-                        w = obj.Base.Shape.Wires[0]
-        if w:
-            if w.isClosed():
+                        baseWire = obj.Base.Shape.Wires[0]
+        if baseWire:
+            if baseWire.isClosed():
                 self.profilsDico = []
                 self.shps = []
-                self.subVolshps = []
+                self.subVolShps = []
                 heights = []
-                edges = Part.__sortEdges__(w.Edges)
-                l = len(edges)
-                for i in range(l):
+                edges = Part.__sortEdges__(baseWire.Edges)
+                if self.flip:
+                    edges = self.flipEdges(edges)
+
+                ln = len(edges)
```

Comment:
In a long method like this one, descriptive names document what you are operating on.
```py
n_edges = len(edges)
```